egs/wsj/s5/steps/cleanup/create_segments_from_ctm_edits.py

```diff
@@ -0,0 +1,549 @@
+#!/bin/env python
+
+# Copyright 2016  Vimal Manohar
+# Apache 2.0
+
+from __future__ import print_function
+import sys, operator, argparse
+import random
+
+# Segment data using CTM and the edits from Levenshtein alignment of 'hypothesis' and 'reference':
+# (i.e. the output of 'align-text' post-processed by 'wer_per_utt_details.pl')
+# This takes as input the file generated by the script
+# steps/cleanup/append_eval_to_ctm.py
+
+# The information added to each token in the CTM is the reference word and one
+# of the following labels:
+#  'C' = correct
+#  'S' = substitution
+#  'D' = deletion
+#  'I' = insertion
+#  'SC' = silence and neighboring hypothesis words are both correct
+#  'SS' = silence and one of the neighboring hypothesis words is a substitution
+#  'SD' = silence and one of the neighboring hypothesis words is a deletion
+#  'SI' = silence and one of the neighboring hypothesis words is an insertion
+#  'XS' = reference word substituted by the OOV symbol in the hypothesis and both the neighboring hypothesis words are correct
+# The priority order for the silence labels is 'SD' > 'SS' > 'SI'.
+
+# Note: Additional lines are added to the CTM to account for deletions.
+
+# Input file:
+# <file-id> <channel> <start-time> <end-time> <conf> <hyp-word> <ref-word> <edit>
+
+## TimBrown_2008P-0007226-0007620 1 0.00 0.10 when <eps> I
+## TimBrown_2008P-0007226-0007620 1 0.10 0.09 i <eps> I
+## TimBrown_2008P-0007226-0007620 1 0.19 0.30 some [NOISE] S
+## TimBrown_2008P-0007226-0007620 1 0.49 0.11 when when C
+## TimBrown_2008P-0007226-0007620 1 0.60 0.06 i i C
+## TimBrown_2008P-0007226-0007620 1 0.66 0.19 say say C
+## TimBrown_2008P-0007226-0007620 1 0.84 0.45 go go C
+## TimBrown_2008P-0007226-0007620 1 1.30 0.31 [COUGH] [COUGH] C
+## TimBrown_2008P-0007226-0007620 1 1.61 0.13 you you C
+## TimBrown_2008P-0007226-0007620 1 1.74 0.00 <eps> [COUGH] D
+## TimBrown_2008P-0007226-0007620 1 1.74 0.00 <eps> a D
+## TimBrown_2008P-0007226-0007620 1 1.74 0.00 <eps> ve D
+## TimBrown_2008P-0007226-0007620 1 1.74 0.18 got got C
+## TimBrown_2008P-0007226-0007620 1 1.92 0.37 thirty thirty C
+## TimBrown_2008P-0007226-0007620 1 2.29 0.83 seconds seconds C
+## TimBrown_2008P-0007226-0007620 1 3.12 0.33 <eps> <eps> SC
+## TimBrown_2008P-0007226-0007620 1 3.45 0.04 [BREATH] [BREATH] C
+## TimBrown_2008P-0007226-0007620 1 3.49 0.11 to to C
+## TimBrown_2008P-0007226-0007620 1 3.60 0.32 [NOISE] [NOISE] C
+
+
+def enum(*sequential, **named):
```

Comment:
I'm going to remove this.. it doesn't seem to be used, but even if it were I wouldn't like it because it's too close to metaprogramming.  It's generally best to work within a language's natural features than to try to emulate the features of other languages.